src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/StreamHelper.cs

```diff
@@ -309,6 +307,14 @@ internal static void WriteToStream(Stream input, Stream output, PSCmdlet cmdlet,
             catch (OperationCanceledException)
             {
             }
+            catch (AggregateException ae)
+            {
+                // we should remove this suppression in next release, which would fix partial file downloads without proper error handling.
+                ae.Handle((x) =>
+                {
+                    return true;
```

Comment:
```suggestion
                    return false;  // Throw all collected exceptions.
```
My apologies.  I was in a hurry and miswrote the code suggestion.  I meant to catch the `AggregateException` here and throw all contained exceptions.  I feel we should not suppress the AggregateException, and instead rethrow the exceptions it contains, and let the upper code layers handle them as they did before.  My guess is that the upper layer did not expect `AggregateException`, but handles other types.

I understand we will not take this change for `7.3` since it is too late in the cycle.  But if/when we take the change for `7.4` I feel this is likely the right way to handle the exceptions.

Reply:
I chatted with @AAATechGuy offline and agree that for `7.4` the aggregateexception catch should be removed.